File: 0153_Find Minimum in Rotated Sorted Array/FindMinimuminRotatedSortedArrayIteration.cs
public class Solution {
    public int FindMin(int[] nums) {
        var l = 0;
        var r = nums.Length-1;
        
        while(l < r)
        {
            var mid = (r-l)/2 + l;
            if(nums[mid] > nums[r])
            {
                l = mid + 1;
            }else{
                r = mid;
            }
        }
        
        return nums[l];
    }
}

File: 0153_Find Minimum in Rotated Sorted Array/FindMinimuminRotatedSortedArrayIteration.java
class Solution {
    public int findMin(int[] nums) {
        int l = 0, r = nums.length-1;
        while(l < r) {
            int mid = (r-l)/2 + l;
            if(nums[mid] < nums[r]) {
                r = mid;
            }else{
                l = mid + 1;
            }
        }
        
        return nums[l];
    }
}